go.mod

```diff
@@ -25,8 +25,8 @@ require (
 	github.com/grpc-ecosystem/grpc-gateway v1.11.1
 	github.com/hashicorp/consul/api v1.2.0 // indirect
 	github.com/jaegertracing/jaeger v1.17.0
+	github.com/james-bebbington/winperfcounters v0.0.0-20200417100953-e7a978334ec0
```

Comment:
I've temporarily put this code into a separate repo: https://github.com/james-bebbington/winperfcounters - it's just a fork of https://github.com/influxdata/telegraf/tree/master/plugins/inputs/win_perf_counters (didn't want to pull in that entire repo as it includes many unrelated dependencies)

I'm not sure where this should live in the long term. One possibility is to try and get this included in `gopsutil`

Reply:
Would try to avoid depending on a "personal" repo.  At least create an issue to track this problem.

Reply:
I would directly depend on telegraf instead. If and when it is included in gopsutil change the dependency. If there is a concern that brings unwanted dependency then this could be moved to the contrib repo. 
Another possibility is to copy the winperfcounters as an internal package. Not sure how licensing would work out.

Reply:
I would also breakdown the commit into two parts to make it easy to review.
1. metrics that depend on runtime (with initial frame work).
2. metrics that depend on winperfcounters

Reply:
> I would directly depend on telegraf instead. If and when it is included in gopsutil change the dependency. If there is a concern that brings unwanted dependency then this could be moved to the contrib repo.

I initially recommended not to depend on telegraf as it brings in too many dependencies. I would have the same concerns if this were in contrib as well. @bogdandrutu @tigrannajaryan  thoughts?

Reply:
> Would try to avoid depending on a "personal" repo. At least create an issue to track this problem.

While I understand the general aversion, we also depend on many libraries in personal repos. It's the nature of git. There isn't much risk given go modules proxying.

Reply:
I'm quite hesitant to pull in telegraf as per jrcamp's reasoning. Given that its only 5-6 relatively small files, I may just bring them into the collector. I've split this PR up now to make it easier to review, but I'll come back to this in the next one.